src/types/topology/routes.rs

```diff
@@ -0,0 +1,10 @@
+use crate::types::topology::{NodeId, NodeRef};
+use std::collections::HashMap;
+
+pub struct Route {
+    //route is the actual vec of references, path is just a "treasure map" with ids
+    path: Vec<NodeRef>,
+}
+pub type Routes = HashMap<NodeId, Route>;
+//Routes are only from client to server and vice versa.
```

Comment:
Unneeded comment in my opinion. If you want to add a constraint you should propose that and add it to the protocol.